django_socio_grpc/protobuf/generation_plugin.py

```diff
@@ -316,3 +331,69 @@ def transform_request_message(
                 service, proto_message, message_name_constructor
             )
         return proto_message
+
+
+@dataclass
+class BaseEnumGenerationPlugin(BaseGenerationPlugin):
+    def transform_request_message(
+        self,
+        service: type["Service"],
+        proto_message: ProtoMessage | str,
+        message_name_constructor: MessageNameConstructor,
+    ) -> ProtoMessage:
+        if isinstance(proto_message, str):
+            return proto_message
+
+        for field in proto_message.fields:
+            if isinstance(field.field_type, type) and issubclass(field.field_type, Enum):
+                self.handle_enum(service, proto_message, field)
+
+        return proto_message
+
+    def transform_response_message(
+        self,
+        service: type["Service"],
+        proto_message: ProtoMessage | str,
+        message_name_constructor: MessageNameConstructor,
+    ) -> ProtoMessage:
+        return self.transform_request_message(service, proto_message, message_name_constructor)
+
+    def handle_enum(
+        self, service: type["Service"], proto_message: ProtoMessage, field: ProtoField
+    ): ...
```

Comment:
maybe add a raise not implemented error with a specific message to tell user that the plugin is not usable as it or at least a warning